googlemaps/places.py

```diff
@@ -0,0 +1,392 @@
+            status: status code
+            results: list of places
+            html_attributions: set of attributions which must be displayed
+
+    """
+    if not (keyword or name or types):
+        raise ValueError("either a keyword, name, or types arg is required")
+
+    return _places(client, "radar", location=location, radius=radius,
+                   keyword=keyword, min_price=min_price, max_price=min_price,
+                   name=name, open_now=open_now, types=types)
+
+
+def _places(client, url_part, query=None, location=None, radius=None,
+            keyword=None, language=None, min_price=0, max_price=4, name=None,
+            open_now=False, rank_by=None, types=None, page_token=None):
+    """
+    Internal handler for ``places``, ``places_nearby``, and ``places_radar``.
+    See each method's docs for arg details.
+    """
+
+    params = {"minprice": min_price, "maxprice": max_price}
+
+    if query:
+        params["query"] = query
+    if location:
+        params["location"] = convert.latlng(location)
+    if radius:
+        params["radius"] = radius
+    if keyword:
+        params["keyword"] = keyword
+    if language:
+        params["language"] = language
+    if name:
+        params["name"] = convert.join_list(" ", name)
+    if open_now:
+        params["opennow"] = "true"
+    if rank_by:
+        params["rankby"] = rank_by
+    if types:
+        params["types"] = convert.join_list("|", types)
+    if page_token:
+        params["pagetoken"] = page_token
+
+    url = "/maps/api/place/%ssearch/json" % url_part
+    return client._get(url, params)
+
+
+def place(client, place_id, language=None):
+    """
+    Comprehensive details for an individual place.
+
+    :param place_id: A textual identifier that uniquely identifies a place,
+                     returned from a Places search.
+    :type place_id: string
+
+    :param language: The language in which to return results.
+    :type langauge: string
+
+    :rtype: result dict with the following keys:
+            status: status code
+            result: dict containing place details
+            html_attributions: set of attributions which must be displayed
+
+    """
+    params = {"placeid": place_id}
+    if language:
+        params["language"] = language
+    return client._get("/maps/api/place/details/json", params)
+
+
+def photo(client, photo_reference, max_width=None, max_height=None):
+    """
+    Downloads a places photo.
+
+    :param photo_reference: A string identifier that uniquely identifies a
+                            photo, as provided by either a Places search or
+                            Places detail request.
+    :type photo_reference: string
+
+    :param max_width: Specifies the maximum desired width, in pixels.
+    :type max_width: int
+
+    :param max_height: Specifies the maximum desired height, in pixels.
+    :type max_height: int
+
+    :rtype: iterator containing the raw image data.
```

Comment:
Great idea, it's really unclear otherwise - done in 3d152eaca4c0f9e51e5bd39140f871b7cc3b3454